pkgs/by-name/do/dorion/package.nix

```diff
@@ -1,65 +1,168 @@
-{ lib
-, stdenv
-, fetchurl
-, autoPatchelfHook
-, dpkg
-, glib-networking
-, gst_all_1
-, libappindicator
-, libayatana-appindicator
-, webkitgtk_4_0
-, wrapGAppsHook3
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchurl,
+  rustPlatform,
+  wrapGAppsHook3,
+  glib-networking,
+  gst_all_1,
+  libayatana-appindicator,
+  nodejs,
+  openssl,
+  pkg-config,
+  yq-go,
+  pnpm,
+  webkitgtk_4_1,
+  copyDesktopItems,
+  cargo-tauri,
+  desktop-file-utils,
 }:
 
+let
+  webkitgtk_4_1' = webkitgtk_4_1.override {
+    enableExperimental = true;
+  };
+
+  shelter = fetchurl {
+    url = "https://raw.githubusercontent.com/uwu/shelter-builds/c6fb071f74089639a7b2b6177144d31192cbf916/shelter.js";
+    hash = "sha256-2zMvRITkgNnLJvKYtdMHwyXKaC8yIwjUezqYmsihz+o=";
+    meta = {
+      homepage = "https://github.com/uwu/shelter";
+      sourceProvenance = [ lib.sourceTypes.binaryBytecode ]; # actually, minified JS
+      license = lib.licenses.cc0;
+    };
+  };
+in
+
+# buildRustPackage doesn't respect cargoRoot
+# ..so use stdenv.mkDerivation and rust hooks instead (see #350541)
 stdenv.mkDerivation (finalAttrs: {
   pname = "dorion";
-  version = "5.0.1";
+  version = "6.2.0";
 
-  src = fetchurl {
-    url = "https://github.com/SpikeHD/Dorion/releases/download/v${finalAttrs.version }/Dorion_${finalAttrs.version}_amd64.deb";
-    hash = "sha256-cCZikTZ+IU3mq/FkJfeggXLyWIsWG+a2qu1GbgW93WQ=";
+  src = fetchFromGitHub {
+    owner = "SpikeHD";
+    repo = "Dorion";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-7isOL/853IMOUkepbu81INNFgjTHpo9dyvPaxMkBU1U=";
   };
 
-  unpackCmd = ''
-    dpkg -X $curSrc .
-  '';
+  cargoRoot = "src-tauri";
+  buildAndTestSubdir = finalAttrs.cargoRoot;
 
-  runtimeDependencies = [
-    glib-networking
-    libappindicator
-    libayatana-appindicator
-  ];
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit (finalAttrs) pname version src;
+    sourceRoot = "${finalAttrs.src.name}/${finalAttrs.cargoRoot}";
+    hash = "sha256-SX6OsGQK8Asj+vXiwK9W/wlYzP7DxRrPmi1EXWQsGQs=";
+    patches = [
+      ./cargo-lock.patch
+    ];
+  };
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-qrB+IpuAhoiKfwZvfF8RSZbiyASGIABmYUd70iqSPyw=";
+  };
 
   nativeBuildInputs = [
-    autoPatchelfHook
-    dpkg
+    pnpm.configHook
+    rustPlatform.cargoSetupHook
+    cargo-tauri.hook
+    rustPlatform.cargoCheckHook
+    nodejs
+    pkg-config
     wrapGAppsHook3
+    yq-go
+    desktop-file-utils
+    copyDesktopItems
   ];
 
   buildInputs = [
-    glib-networking
+    openssl
+    webkitgtk_4_1
```

Comment:
Some of these dependencies should be optional if this app supports both Linux and Darwin